src/Core/src/AbstractApi.php

```diff
@@ -117,7 +118,11 @@ public function __construct($configuration = [], ?CredentialProvider $credential
             $request->setHeader('content-length', $length);
         }
 
-        return $this->httpClient->request($request->getMethod(), $request->getUrl(), ['headers' => $request->getHeaders(), 'body' => $request->getBody()]);
+        if (($body = $request->getBody()) instanceof StringStream) {
+            $body = $body->stringify();
+        }
```

Comment:
So if the $body is a CallabeStream... then we pass that object to the `httpClient`, does that work?

Reply:
Oh, sorry. My bad. 

A `Stream` is a `IteratorAggregate` which is Traversable. https://3v4l.org/Og6Bb

Reply:
Yes, because a stream is an `IteratorAggregate` which is handled by `HttpClient`: It yield on the iterator and send chunks. 

Issue is: some Docker Image does not support receiving chunked content. We didn't noticed it before, because it looks like php-curl sometime does not send the in chunks...

Please don't merge it yet, I'll try to use `transfert-encoding: identity`

Reply:
> Issue is: some Docker Image does not support receiving chunked content. 

Please add that as a comment =)

Reply:
ouch.. in fact, there were a huge bug...

Because of Iterable body, `curl` converts the HEAD/GET method in PUT which break Rest APIs  

Please have a second review.

Reply:
Why does curl do that?

Reply:
Not sure, probably related to the way symfony/httpclient set the request body (`CURLOPT_READFUNCTION`)